.changeset/stale-ducks-protect.md

```diff
@@ -0,0 +1,5 @@
+---
+"astro": patch
+---
+
+Fixes an issue that causes static entrypoints build to fail when the path contains the extension several times. For instance, `./.astro/index.astro` caused such issue.
```

Comment:
```suggestion
Fixes an issue that causes static entrypoints build to fail when they are saved in folders that contain `.astro` in their name. For instance, `./.astro/index.astro` caused such issue.
```

Let me know what you think of this suggestion.

Reply:
Actually I found that this was true for any extension, eg. `./.mdx/index.mdx` etc so not specifically tied to `.astro`

Reply:
I see, so any dot file/folder

Reply:
```suggestion
Fixes an issue that causes static entrypoints build to fail because of the path in certain conditions. Specifically, it failed if the path had an extension (like `.astro`, `.mdx` etc) and such extension would be also within the path (like `./.astro/index.astro`).
```
What do you think about this one?